class-tgm-plugin-activation.php

```diff
@@ -1128,6 +1108,75 @@ public function _get_plugin_data_from_name( $name, $data = 'slug' ) {
 		}
 
 		/**
+		 * Retrieve the download url for a package.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param string $slug
+		 *
+		 * @return string Plugin download URL or path to local file
+		 */
+		public function get_download_url( $slug ) {
+			$dl = '';
+
+			if ( ! empty( $this->plugins[ $slug ]['source'] ) ) {
+
+				$source = $this->plugins[ $slug ]['source'];
+
+				// Is this a WP repo plugin ?
+				if ( 'repo' === $source || preg_match( self::WP_REPO_REGEX, $source ) ) {
```

Comment:
I see it's [not](https://github.com/thomasgriffin/TGM-Plugin-Activation/pull/374/files#diff-6f3dff769e44625dd72b031675006eb3R1115), but as before, these conditions could be wrapped in boolean methods so there is only one instance of each.